Data Access Usage Example

philipc2 opened the pull request:
Usage Example showcasing different ways of accessing data stored in `Grid` objects. Notebook only highlights variables related to grid files, will add examples once the design of `open_dataset()` with data files is implemented. 

1. Indexing with Original Name 
2. UGRID access attributes
3. Indexing with Variable Dictionary

https://uxarray--137.org.readthedocs.build/en/173/examples.html

philipc2 commented:
@erogluorhan 

Any comments on the general structured of the notebook before I add any text cells / background info?

erogluorhan commented:
> @erogluorhan
> 
> Any comments on the general structured of the notebook before I add any text cells / background info?

I cannot see your notebook in the rendered docs. I can't see Anissa's examples page improvements either.

erogluorhan commented:
> > > @erogluorhan
> > > Any comments on the general structured of the notebook before I add any text cells / background info?
> > 
> > 
> > I cannot see your notebook in the rendered docs. I can't see Anissa's examples page improvements either.
> 
> Should be fixed now!

Docs generation is failing; please fix.

anissa111 commented:
@philipc2 I've added the example to the gallery.yml file, that should make it show up on the examples page!

erogluorhan commented on docs/examples/data-access.ipynb:
```diff
@@ -0,0 +1,1042 @@
+   "source": [
+    "# Accessing Data\n",
+    "\n",
+    "There are many different conventions used to define unstructured grids (UGRID, SCRIP, etc.). These conventions often have different variable names used to describe the same variables.\n",
```
Slight correction: "There are many different grid types ..."

Remove: "These conventions often have different variable names used to describe the same variables"

erogluorhan commented on docs/examples/data-access.ipynb:
```diff
@@ -0,0 +1,1042 @@
+    "ugrid_grid_path = base_path + \"outCSne30.ug\"\n",
+    "\n",
+    "# Non-UGRID Grid File\n",
+    "non_ugrid_grid_path = base_path + \"geoflow-small/grid.nc\""
```
This is actually a UGRID file with different variable namings. We might need to modify the entire structure of the notebook.

erogluorhan commented on docs/examples/data-access.ipynb:
```diff
@@ -0,0 +1,1042 @@
+    "\n",
+    "There are many different conventions used to define unstructured grids (UGRID, SCRIP, etc.). These conventions often have different variable names used to describe the same variables.\n",
+    "\n",
+    "UXarray offers multiple ways of accessing data stored in `Grid` objects. This notebook will highlight different ways of accessing data with examples both in UGRID and non-UGRID formats.\n",
```
Tell something about the facts that 

(1) whatever the grid type UXarray reads in, it converts it to a UGRID type in the Grid object.  
(2) The UGRID convention is flexible about variable namings except a few required topology attributes (such as "cf_role", "topology_dimension", etc.)
(3) To handle all of those variable names coming from either non-UGRID or UGRID types, UXarray Grid object makes use of a dictionary in which it uses a set of standardized names for keys and custom names as values.
      - There are a few different ways to access those variables in UXarray Grid object.

erogluorhan reviewed (changes requested):
I've put a few inline comments. We may need to play with the entire structure of the notebook.  I am happy to walk you through my thoughts in a short meeting if you want.

philipc2 commented:
@erogluorhan @rajeeja 

New notebook is pushed. Addressed most of the comments from our meeting last week. Let me know if this new layout works better. 

I'm also considering adding a short blurb about UGRID conventions with a diagram, would that be a good addition?

rajeeja commented:
😊 diagrams, yay!

erogluorhan commented:
Close in favor of PR #164
